src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts

```diff
@@ -362,6 +362,54 @@ describe('ProgrammingExerciseConfigureGradingComponent', () => {
         flush();
     }));
 
+    it('should show error alert when test case weights are less or equal zero', fakeAsync(() => {
+        initGradingComponent({ showInactive: true });
+
+        fixture.detectChanges();
+
+        const orderedTests = _sortBy(testCases1, 'testName');
+
+        const table = debugElement.query(By.css(testCaseTableId));
+
+        // get all input fields
+        const editingInputs = table.queryAll(By.css(tableEditingInput));
+        expect(editingInputs).to.have.lengthOf(testCases1.length * 3);
+        // Set only the weight input fields to 0 of all test cases
+        for (let i = 0; i < editingInputs.length; i += 3) {
+            const weightInput = editingInputs[i].nativeElement;
+            expect(weightInput).to.exist;
+            weightInput.focus();
+
+            // Set new weight.
+            weightInput.value = '0';
+            weightInput.dispatchEvent(new Event('blur'));
+        }
+
+        fixture.detectChanges();
+        expect(comp.changedTestCaseIds).to.deep.equal(orderedTests.map((test) => test.id));
+
+        // Mock which should be return from update service call
+        const updateTestCases = orderedTests.map((test) => {
+            return { ...test, weight: 0, bonusMultiplier: 2, bonusPoints: 1 };
+        });
+        // Save weight.
+        updateTestCasesStub.returns(of(updateTestCases));
+
+        // Initialize spy for error alert
+        const alertService = TestBed.inject(JhiAlertService);
+        const alertServiceSpy = spy(alertService, 'error');
+
+        const saveButton = getSaveButton();
+        expectElementToBeEnabled(saveButton);
+        saveButton.click();
+
+        expect(alertServiceSpy).to.be.calledOnce;
+
+        tick();
+        fixture.destroy();
```

Comment:
This and the ticks seems not needed. I removed them and it still works. The component is recreated in the beforeEach already

Reply:
Done in `8177c86`